### DISCUSSION TITLE
Can not connect to Influxdb V2

### DISCUSSION THREAD

**Discussion by PietFourie** [johanmeijer/grott#465] [Q&A]
HI
I setup a docker system Grott/MQTT/Telegraf/inluxdb/Grafana.
All are communicating well on their own network and the data is stored in the influxdb.
However, there is no proper telegraf plugin and the data is not properly stored. I have 2 inverters and the data does not tag the "devices" properly. The fields are not what is transmitted.

I then decided to allow grott to write directly to influxdb, using the grott.ini option.
Communication is received by influxdb but it is rejected via message.....

Grott records whitelisted :  ['0103', '0104', '0116', '0119', '0150', '5003', '5004', '5016', '5019', '5050', '5103', '5104', '5116', '5119', '5150']
	 - Grott InfluxDB V2 initiating started
	 - Grott error: can not contact InfluxDB
(401)
Reason: Unauthorized
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json; charset=utf-8', 'X-Influxdb-Build': 'OSS', 'X-Influxdb-Version': '2.0.9', 'X-Platform-Error-Code': 'unauthorized', 'Date': 'Tue, 24 Oct 2023 08:08:11 GMT', 'Content-Length': '55'})
HTTP response body: {"code":"unauthorized","message":"unauthorized access"}
Grott Influxdb initialisation error

All credentials are correct and are used in Telegraph.
Somehow grott does not send the credentials correctly to influxdb V2
... or so I believe.
My grott.ini section
influx = True       
influx2 = True      
dbname = "Grott2"
ip = 172.20.0.2
port = 8086
user = XXXXXX
password = xxxxxxx
token  = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
org  = "Growatt_Home"
bucket = "Grott2" 

Any help[ would be highly appreciated. 
Also some guide to use telegraf plugin, which one, etc? maybe setting "tags"?

I am new to this.......

**Comment by johanmeijer** [comment]
Ok. It is already a while ago when I created the Grott influxdb2 interface. 

One think for sure there are people using this interface successfully (including my test environment, that I have to set up again to test).  
So I think it should work (and Grott passes the credential correctly). 

At the wiki (https://github.com/johanmeijer/grott/wiki/InfluxDB-Support) I documented (and again I have to retest it myself) : 

[influx]
influx = true
influx2 = true
ip = localhost
port = 8086
token  = influx_token
org  = your_org
bucket = your_bucket 

Is it possible that you have specified to much (it should not hurt but you never know......) ?

If you specify mqttinverterintopic = True in the MQTT section of the grott.ini there will be different topics per inverter created,

**Comment by PietFourie** [comment]
Hi
Before I start ... Thank you for this incredible software that you have written. One can see it took great effort.

Let me start by saying I am new to this thing. I tried to set up docker images of grott, mqtt, telegraf, influxdb and grafana. They all work on their own network and communication is correct. grott send the data to mqtt, mqtt sends it to telegraph and telegraf to influxdb. I can see the data in MQTT Explorer and the influxdb UI. However, in influxdb the fields and measurements are not all there and I can not select or distinguish the two inverters. I connected using influxQL and Flux protocols. I even set the retention policy to use QL . However you can not select the inverter. It is just not an option.
I then tried to use the influxdb option in grott for direct writing, cutting out mqtt and telegraf. but I get the unauthorised response.
Then I tried to use the extension grotcsv but I do not know how to load that into influxdb. It works from grott's side. The influxdb requires some settings for the headers, etc. Somehow one must use telegraf to enter it into influxdb but at this stage I do not how.
Then I wanted to write telegraf pllugin but it is written in GO... damm!
So to sum it up grott, mqtt, telegraf, inlfudx path works but it does not process the correct data. The TAG just does not show up. All credentials for token, org and bucket are used the same over the config files.
My best option is now the direct writing of influxdb from grott or go back the mqtt route.
I used vscode to debug run your code. 

(grottconf.py: line 165)

              #self.influxclient = InfluxDBClient(url='192.168.0.211:8086',org=self.iforg, token=self.iftoken)
                self.influxclient = InfluxDBClient(url="{}:{}".format(self.ifip, self.ifport),org=self.iforg, token=self.iftoken)
                self.ifbucket_api = self.influxclient.buckets_api()
                self.iforganization_api = self.influxclient.organizations_api()              
                self.ifwrite_api = self.influxclient.write_api(write_options=SYNCHRONOUS)  

                try:
                    buckets = self.ifbucket_api.find_bucket_by_name(self.ifbucket)
 It breaks here ... and returns error.

I run Linux mint 20.3 and InfluxDB-oss 2.0.9 and 2.7.1
If I can just write a telegraf plugin to process the mqtt data .......

**Reply by anoppe** [reply]
what happens if you remove the quotes `"` from the token in your config? (the same holds for the other config enties)

**Reply by PietFourie** [reply]
Nnnnooooooooo.............
It worked!
I did remove the quotes in my various trials but it seems I did something wrong then...
To sum it up.... No quotes around token, bucket and org.
Problem solved. 
Thanks a lot ....

**Comment by PietFourie** [comment]
solved
